server/pkg/asset/preview_type.go

```diff
@@ -44,6 +47,8 @@ func PreviewTypeFromRef(p *string) *PreviewType {
 		p2 = PreviewTypeGEO3D
 	case PreviewTypeMODEL3D:
 		p2 = PreviewTypeMODEL3D
+	case PreviewTypeUNKNOWN:
+		p2 = PreviewTypeUNKNOWN
 	default:
 		return nil
```

Comment:
Implementation of this func is duplicated with `PreviewTypeFrom`.  Use `PreviewTypeFrom(*p)` instead.

Reply:
Done!

Reply:
Do not forget fixing it. Implementation duplicated.

Reply:
I am not sure if I get what you mean, can you please explain when you have time. Thank you!

Reply:
Done!